vault/secret_test.go

```diff
@@ -1,19 +1,90 @@
 package vault
 
-import "testing"
+import (
+	"testing"
+	"math/rand"
+	"time"
+)
 
-const secretName = "/test/secret"
+const letterBytes = "abcdefghijklmnopqrstuvwxyz"
+
+func init() {
+    rand.Seed(time.Now().UnixNano())
```

Comment:
@andrii-zakurenyi are you ok with this here? I'd prefer we avoid `init` with random and put in the function itself since we shuffle tests and want to support parallel tests if required. Do you see a need for this outside the function?

Reply:
Let me know how you want to proceed with this and I'll update the PR accordingly.

Reply:
@michaelsauter awesome. Appreciate the effort and quick response. Is there a reason we need to seed it vs just using random in the func itself? We can always adjust later, but let me just confirm that with you and we can try to get this merged in quick. 

Cheers

Reply:
It is in tests, so yeah, I'm totally ok.

Also since Go 1.20:
> The [math/rand](https://go.dev/pkg/math/rand/) package now automatically seeds the global random number generator (used by top-level functions like Float64 and Int) with a random value, and the top-level [Seed](https://go.dev/pkg/math/rand/#Seed) function has been deprecated.

:man_shrugging:

Reply:
@andrii-zakurenyi Thanks for the pointer! I missed that change in 1.20.

I was about to remove `rand.Seed` but then saw that `go.mod` has `go 1.18` still, and I'd keep the version in `go.mod` consistent with the code, even though it is not technically required (yet). 

Maybe best to create a separate ticket that updates `go.mod` and removes the call to `rand.Seed`?

Reply:
We can change that separately. Let me approve so we get it done. Release/tag will be done separately. Much appreciate your contribution 🎉